lessons/advanced/typespec.md

```diff
@@ -13,17 +13,17 @@ In this lesson we will learn about `@spec` and `@type` syntax. First is more syn
 
 ## Introduction 
 
-It's not uncommon you would like to describe interface of your function. You could use [@doc annotation](../../basics/documentation), but it is only information for other developers that is not checked in compilation time. For this purpose Elixir has `@spec` annotation to describe specification of function that will be checked by compiler.
+It's not uncommon you would like to describe interface of your function. You could use [@doc annotation](../../basics/documentation), but it only serves as documentation for other developers. It is not checked at compilation time. For this purpose Elixir has `@spec` annotation to describe specification of function that will be checked by compiler.
 
-However in some cases specification is going to be quite big and complicated. If you would like to reduce complexity, you want to introduce custom type definition. Elixir has `@type` annotation for that. In the other hand, Elixir is still dynamic language. That means all information about type will be ignored by compiler, but could be used by other tools.   
+However in some cases specification is going to be quite big and complicated. If you would like to reduce complexity, you want to introduce custom type definition. Elixir has `@type` annotation for that. In the other hand, Elixir is still a dynamic language. That means all information about types will be ignored by compiler, but could be used by other tools.   
```

Comment:
If `However` is used as a conjunctive adverb in this case, shouldn't it be followed by a comma?